Communications/Packet.cs

```diff
@@ -35,6 +35,18 @@ public Packet(byte ID, bool IsUDP, string Endpoint = null)
         /// <param name="Data"> Data to append to packet. </param>
         public void AppendData(byte[] NewData) { this.Data.AppendData(NewData); }
 
+
+        /// <summary> Get a part of current data. </summary>
+        /// <param name="Offset"> Index to start copying. </param>
+        /// <param name="Size"> Size of data to copy. </param>
+        /// <returns> Data with index in range [Offset, Offset + Size). </returns>
+        public byte[] GetDataSlice(int Offset, int Size) { return this.Data.GetDataSlice(Offset, Size); }
+
+        /// <summary> Get a part of current data from an offset to the end. </summary>
+        /// <param name="Offset"> Index to start copying. </param>
+        /// <returns> Data from `Offset` to the end. </returns>
+        public byte[] GetDataSlice(int Offset) { return this.Data.GetDataSlice(Offset); }
+
```

Comment:
We should probably have a discussion about whether or not GetDataSlice() and even AppendData() makes sense to have in Packet, since all it does is reference the Data field and call the same method. Personally, if we really want to separate Data and the Comms information, I don't think that these methods have a place in Packet.

Reply:
That is a good point. I agree that we should separate the job of of `Packet` class and `Message` class.  I may revise this part later (maybe tomorrow?).